### PR TITLE
Support for Listing Services for Node and Retrieving Nodes that support Mesh Capable Service

### LINKED ISSUE #272: Support for Listing Services for Node and Retrieving Nodes that support Mesh Capable Service
### Is your feature request related to a problem? Please describe.

no

### Describe the solution you'd like

add support for this https://developer.hashicorp.com/consul/api-docs/catalog#list-services-for-node
and https://developer.hashicorp.com/consul/api-docs/catalog#list-nodes-for-mesh-capable-service
This API is supported in the golang client (1.8.19)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
fixes #272
@marcin-krystianc @naskio

### CODE REVIEW

File: Consul/Catalog.cs [hunk 1]
```diff
@@ -237,6 +272,44 @@ public Task<QueryResult<CatalogService[]>> Service(string service, string tag, Q
...
+        /// Returns the nodes providing a mesh-capable service in a given datacenter.
+        /// </summary>
+        /// <param name="service">The service ID</param>
+        /// <param name="q">Customized Query options</param>
+        /// <param name="filter">Specifies the expression used to filter the queries results prior to returning the data</param>
+        /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
+        /// <returns>A list of service instances</returns>
+        public Task<QueryResult<CatalogService[]>> NodesForMeshCapableService(string service, QueryOptions q, Filter filter, CancellationToken ct = default)
+        {
+            var req = _client.Get<CatalogService[]>(string.Format("/v1/catalog/connect/{0}", service), q, filter);
```
Comment (marcin-krystianc) [inline Consul/Catalog.cs:308]: I don't think we need the temporary `req` variable here? We don't use it in other similar places.

File: Consul.Test/CatalogTest.cs [hunk 2]
```diff
@@ -174,6 +175,31 @@ public async Task Catalog_GetTaggedAddressesService()
...
+        public async Task Catalog_GetNodesForMeshCapableService()
+        {
+            var svcID = KVTest.GenerateTestKeyName();
+            var registration = new AgentServiceRegistration
+            {
+                Name = svcID,
+                Port = 8000,
+                Connect = new AgentServiceConnect
+                {
+
```
Comment (marcin-krystianc) [inline Consul.Test/CatalogTest.cs:188]: nit: extra new line

File: Consul.Test/CatalogTest.cs [hunk 3]
```diff
@@ -238,5 +264,54 @@ public async Task Catalog_EnableTagOverride()
                 Assert.True(services.Response[0].ServiceEnableTagOverride);
             }
         }
+
+        [Fact]
+        public async Task Catalog_ServicesForNodes()
```
Comment (marcin-krystianc) [inline Consul.Test/CatalogTest.cs:269]: See here (https://github.com/G-Research/consuldotnet/blob/5da7a42135bfa4e2352a57a32bbed4172324abe4/Consul.Test/AgentTest.cs#L543-L547), how to make the skip versions which do not support tested API.

File: Consul/Catalog.cs [hunk 4]
```diff
@@ -84,6 +85,40 @@ public class CatalogDeregistration
...
+        public Dictionary<string, ServiceAddress> TaggedAddresses { get; set; }
+    }
+
+    public class ServiceAddress
+    {
+        public string Address { get; set; }
+        public int Port { get; set; }
+    }
+
+
```
Comment (marcin-krystianc) [inline Consul/Catalog.cs:121]: nit: extra new line

**Review by marcin-krystianc — COMMENTED** [review]
Just few minor comments, looks good in general.

File: Consul.Test/CatalogTest.cs (hunk 3 above)
Comment (sammychinedu2ky) [reply Consul.Test/CatalogTest.cs:269]: alright thanks @marcin 🚀🚀. Really learning a lot

File: Consul.Test/CatalogTest.cs [hunk 5]
```diff
@@ -238,5 +265,57 @@ public async Task Catalog_EnableTagOverride()
                 Assert.True(services.Response[0].ServiceEnableTagOverride);
             }
         }
+
+        [SkippableFact]
+        public async Task Catalog_ServicesForNodes()
+        {
+            var cutOffVersion = SemanticVersion.Parse("1.7.0");
+            Skip.If(AgentVersion < cutOffVersion, $"Current version is {AgentVersion}, but `logjson` is only supported from Consul {cutOffVersion}");
```
Comment (marcin-krystianc) [inline Consul.Test/CatalogTest.cs:273]: Please update the message to mention the actual feature which is not supported (instead of `logjson`)

Comment (sammychinedu2ky) [reply Consul.Test/CatalogTest.cs:273]: ooh my bad

**Comment by sammychinedu2ky** [comment]
done @marcin-krystianc 🚀. thank you

**Pull request merged**